### PR TITLE
[Console] respect multi-byte characters when rendering vertical-style tables

### LINKED ISSUE #54433: Wrong formatting of vertical style tables with headers containing multi-byte characters
### Symfony version(s) affected

6.4.2

### Description

When using _vertical style tables_, whenever the component has to calculate the header width dynamically, the calculated width uses the total number of bytes of the string involved instead of the actual number of characters that would be rendered. That results in misaligned columns (and ugly tables) if the length of actual renderable characters is smaller than the next smallest length.

### How to reproduce

Please note that I am using a Laravel console command for **convenience only**. The code that matters uses **only** Symfony objects and methods!

Create and run the following code inside any Laravel project. The console command may be created using ```php artisan make:command TestBrokenTable```. _Again, this is only by convenience, the affected code does not depend on Laravel!_ (I do not have any Symfony framework project available!)

```php
<?php

namespace App\Console\Commands;

use Illuminate\Console\Command;

use Symfony\Component\Console\Helper\Table;
use Symfony\Component\Console\Output\BufferedOutput;

class TestBrokenTable extends Command
{
	/**
	 * The name and signature of the console command.
	 *
	 * @var string
	 */
	protected $signature = "app:test-broken-table";

	/**
	 * The console command description.
	 *
	 * @var string
	 */
	protected $description = "Command description";

	/**
	 * Execute the console command.
	 */
	public function handle()
	{
		$headers = [
			"long without MB chars",
			"áéíóú MB chars",					// <-- This will cause problems
			"Column 3 anything",
		];

		// This will NOT cause problems...

		// $headers = [
		// 	"long without MB chars",
		// 	"aeiou MB chars",
		// 	"Column 3 anything",
		// ];

		$this->line("Middle Header:");
		$this->line("Bytes: ".strlen($headers[1]));
		$this->line("Chars: ".mb_strlen($headers[1]));
		
		$data = [
			[
				"Shorter value",
				"Value with MB chars áéíóú",
				"Another Value",
			],
		];
		
		$output = new BufferedOutput(decorated: true);
		
		$table = new Table($output);
		$table->setHeaders($headers);
		$table->setRows($data);
		$table->setVertical();
		
		$table->render();
		
		$text = $output->fetch();
		
		print($text);
	}
}

```

The output is as follows:

image

If I remove the multi-byte characters, the output is now correct:

image

If I use a table with horizontal (standard) layout, the problem does not occur.

### Possible Solution

_No response_

### Additional Context

All of the strings are in UTF-8 format.
Using Microsoft Visual Studio Code (latest update as of 2024-03-28).
Using Windows 10 Professional (latest update as of 2024-03-28).

In the above examples, "bytes" are the result of the ```strlen()``` function and "chars" the result of ```mb_strlen()```.

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 6.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Issues        | Fix #54433
| License       | MIT

### CODE REVIEW

**Review by derrabus — APPROVED** [review]

File: src/Symfony/Component/Console/Tests/Helper/TableTest.php [hunk 1]
```diff
@@ -1649,6 +1649,28 @@ public static function provideRenderVerticalTests(): \Traversable
             $books,
         ];
 
+        yield 'With multibyte characters in some cells' => [
```
Comment (stof) [inline src/Symfony/Component/Console/Tests/Helper/TableTest.php]: Just to confirm, where are the multibytes characters in the headers ?

And the tests should be renamed: your fix is about characters in headers

Comment (xabbuh) [reply src/Symfony/Component/Console/Tests/Helper/TableTest.php]: it's the `í` in `Dívíne` (I have clarified that in the fixture when renaming it)

File: src/Symfony/Component/Console/Tests/Helper/TableTest.php [hunk 2]
```diff
@@ -1649,6 +1649,28 @@ public static function provideRenderVerticalTests(): \Traversable
             $books,
         ];
 
+        yield 'With multibyte characters in some headers (the "í" in "Dívíne")' => [
```
Comment (stof) [inline src/Symfony/Component/Console/Tests/Helper/TableTest.php]: This is not reproducing the bug as `Dívíne` is in the value, not in the header.

**Review by stof — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Console/Tests/Helper/TableTest.php (hunk 2 above)
Comment (derrabus) [reply src/Symfony/Component/Console/Tests/Helper/TableTest.php]: But the `í` in `Títle` is in the header, isn't it?

Comment (xabbuh) [reply src/Symfony/Component/Console/Tests/Helper/TableTest.php]: Ah yes, you are right. I’ll update the fixture name to „Títle“ as Alexander figured correctly.

Comment (xabbuh) [reply src/Symfony/Component/Console/Tests/Helper/TableTest.php]: should finally be fixed now

**Review by derrabus — APPROVED** [review]

**Review by stof — APPROVED** [review]

**Review by chalasr — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**